tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -78,32 +80,48 @@ def sync(sync_ttl):
                                     integration_index_key,
                                     integration_id
                                 )
-                                # Let other nodes sync up with
-                                # integration_id
-                                params = {
-                                    'TendrlContext.integration_id':
-                                    integration_id,
-                                }
-                                payload = {
-                                    "tags": [_target],
-                                    "run": _flow,
-                                    "status": "new",
-                                    "parameters": params,
-                                    "type": "node"
-                                }
-                                _job_id = str(uuid.uuid4())
-                                _job = Job(
-                                    job_id=_job_id,
-                                    status="new",
-                                    payload=payload,
-                                    timeout="no"
-                                ).save()
-                                while True:
-                                    _job = _job.load()
-                                    if _job.status in ["finished",
-                                                       "failed"]:
-                                        break
-
+                                # Raise an alert regarding the same
+                                msg = "New peers identified in cluster: %s. " \
+                                    "Make sure tendrl-ansible is executed for " \
+                                    "the new nodes so that expand cluster " \
+                                    "option can be triggered" % integration_id
+                                event_utils.emit_event(
+                                    "cluster_status",
+                                    "expand_pending",
+                                    msg,
+                                    "cluster_{0}".format(integration_id),
+                                    "WARNING",
+                                    integration_id=integration_id
+                                )
+                            nodes_ids = json.loads(NS._int.client.read(
+                                "indexes/tags/tendrl/integration/%s" % integration_id
+                            ).value)
+                            cluster_peers = NS._int.client.read(
+                                "/clusters/%s/Peers" % integration_id
```

Comment:
NS.gluster.objects.Peer is not available unless tendrl-gluster-intergration is running.  Put this in a try catch and also use etcd_utils everywhere in your code